parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcess.java

```diff
@@ -156,63 +160,66 @@ private void doWork() {
             for (File from : reports) {
-                ReportParserDescriptor rpd = getReportParserDescriptor(from);
-                if (rpd != null) {
+                List<ReportParserDescriptor> rpds = getReportParserDescriptors(from);
+                if(rpds.isEmpty()) {
+                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
+                    continue;
+                }
+                for(ReportParserDescriptor rpd : rpds) {
                     _build.getBuildLogger().message("Transforming "+from.getAbsolutePath()+" with "+rpd.getLabel());
-                    String targetFileName = PREFIX + from.getName();
+                    String targetFileName = rpd.getOutputFileNamePrefix() + from.getName();
                     File to = new File(from.getParentFile(), targetFileName);
                     transform(from, to, rpd.getXSL(), checkoutDir);
-                } else {
-                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
                 }
             }
         }
     }
 
-    private ReportParserDescriptor getReportParserDescriptor(File from) {
-        StreamSource xml = new StreamSource(from);
-        XMLEventReader reader = null;
+    private List<ReportParserDescriptor> getReportParserDescriptors(File from) {
+        List<ReportParserDescriptor> descriptors = new ArrayList<ReportParserDescriptor>();
         try {
-            reader = xmlInputFactory.createXMLEventReader(xml);
-            while (reader.hasNext()) {
-                XMLEvent event = reader.nextEvent();
-                switch (event.getEventType()) {
-                case XMLStreamConstants.START_ELEMENT:
-                    String toolName = null;
-                    StartElement startElement = event.asStartElement();
-                    @SuppressWarnings("unchecked")
-                    Iterator<Attribute> attributes = startElement.getAttributes();
-                    while (attributes.hasNext()) {
-                        Attribute attribute = attributes.next();
-                        String name = attribute.getName().getLocalPart();
-                        if ("toolName".equals(name)) {
-                            toolName = attribute.getValue();
-                            break;
-                        }
-                    }
-                    if (toolName != null) {
-                        return "SOAtest".equals(toolName) ?
-                                ReportParserTypes.getDescriptor(ReportParserType.SOATEST.name()) :
-                                ReportParserTypes.getDescriptor(ReportParserType.ANALYZERS.name());
-                    }
-                    return null;
-                default:
-                    break;
-                }
+            Document document = getDocument(from);
+            if(checkIfNodeExists(document, "/ResultsSession/CodingStandards/StdViols/StdViol")) {
```

Comment:
If I didn't get it wrong, unit test results and static analysis results could be generated by one tool in one report file, but SOAtest test results will be in separate file. If that's the case, when the report is a SOAtest report, it should not need to check for other descriptors. I think that's the reason why the original logic is only picking one of the descriptor.